x-pack/filebeat/input/gcppubsub/input.go

```diff
@@ -99,6 +94,9 @@ func NewInput(
 		}
 	}()
 
+	metrics := newInputMetrics(conf.ProjectID, nil)
```

Comment:
since this input is v1 and not v2, I don't have a context id to use here (like we do for the other inputs), so I went with the projectId from the config

Reply:
You can also make an ID from hashing the config.
```
func configID(config *conf.C) (string, error) {
	var tmp struct {
		ID string `config:"id"`
	}
	if err := config.Unpack(&tmp); err != nil {
		return "", fmt.Errorf("error extracting ID: %w", err)
	}
	if tmp.ID != "" {
		return tmp.ID, nil
	}

	var h map[string]interface{}
	_ = config.Unpack(&h)
	id, err := hashstructure.Hash(h, nil)
	if err != nil {
		return "", fmt.Errorf("can not compute ID from configuration: %w", err)
	}

	return fmt.Sprintf("%16X", id), nil
}
```